lib/widgets-page.php

```diff
@@ -11,10 +11,21 @@
  * @since 5.2.0
  */
 function the_gutenberg_widgets() {
+	/** This action is documented in wp-admin/admin-footer.php */
+	// phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
+	do_action( 'admin_print_styles-widgets.php' );
+	// phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
+	do_action( 'admin_print_scripts-widgets.php' );
 	?>
 	<div id="widgets-editor" class="blocks-widgets-container">
 	</div>
 	<?php
+	/** This action is documented in wp-admin/admin-footer.php */
+	// phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
+	do_action( 'admin_footer-widgets.php' );
+	/** This action is documented in wp-admin/admin-footer.php */
+	// phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
+	do_action( 'admin_print_footer_scripts-widgets.php' );
```

Comment:
Performing these actions here means that the style and script tags are located in the body of the page instead of in the header and footer. We should call `do_action()` within the `admin_print_styles`, `admin_print_scripts`, `admin_footer`, and `admin_footer_scripts` hooks.

I'm thinking this simplest way to do this is to rely on `widgets.php` to make these `do_action` calls and change the `get_current_screen()->is_block_editor()` checks in `widgets.php` to use our own version of `is_block_editor()`.

```php
 function gutenberg_is_block_editor() {
	$screen = get_current_screen();
	return $screen->is_block_editor() || 'gutenberg_page_gutenberg-widgets' === $screen->id;
 }
```